dev/helpers/mssql/run_migrations.sh

```diff
@@ -75,12 +80,32 @@ migrate () {
   /opt/mssql-tools/bin/sqlcmd -S $SERVER -d $DATABASE -U $USER -P $PASSWD -I -i $file
 }
 
-for f in `ls -v $MIGRATE_DIRECTORY/*.sql`; do
-  BASENAME=$(basename $f)
-  if should_migrate $f == 1 ; then
-    migrate $f
-    record_migration $f
-  else
-    echo "Skipping $f, $BASENAME"
+
+if [[ "$RERUN_SCRIPT_NAME" != "" ]]; then
+  full_migration_path=$MIGRATE_DIRECTORY/$RERUN_SCRIPT_NAME
+  if [[ -f $full_migration_path ]]; then
+    query="SELECT * FROM [migrations] WHERE [Filename] = '$RERUN_SCRIPT_NAME'"
+    result=$(/opt/mssql-tools/bin/sqlcmd -S $SERVER -d migrations_$DATABASE -U $USER -P $PASSWD -I -Q "$query")
+
+    if [[ "$result" =~ .*$RERUN_SCRIPT_NAME.* ]]; then
```

Comment:
I think I'd replace this with a check if `should_migrate == 0`, that way you don't need to repeat all the should_migrate logic.

Reply:
Or you can just always run the delete

Reply:
Thanks for catching this!

I'm a bit confused with how the script is currently working as it seems that my would have to be replaced with `should_migrate == 1` for the logic to match. I tested this code and it seemed to work as expected, but it seems like the opposite is working later on when running all of the migrations that haven't run. This will take more testing on my part to fix.